pythonPackages.geojson: 2.5.0 -> 3.0.0

oxzi opened the pull request:
###### Description of changes
Version bump, https://github.com/jazzband/geojson/blob/3.0.0/CHANGELOG.rst#300-2022-01-26

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

oxzi committed:
pythonPackages.geojson: 2.5.0 -> 3.0.0

https://github.com/jazzband/geojson/blob/3.0.0/CHANGELOG.rst#300-2022-01-26

mattmelling commented:
Result of `nixpkgs-review pr 214533` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages failed to build:</summary>
  <ul>
    <li>home-assistant-component-tests.geo_json_events</li>
    <li>home-assistant-component-tests.openweathermap</li>
    <li>python310Packages.aio-geojson-generic-client</li>
    <li>python310Packages.pyowm</li>
    <li>python311Packages.agate-sql</li>
    <li>python311Packages.aio-geojson-client</li>
    <li>python311Packages.aio-geojson-generic-client</li>
    <li>python311Packages.aio-geojson-geonetnz-quakes</li>
    <li>python311Packages.aio-geojson-geonetnz-volcano</li>
    <li>python311Packages.aio-geojson-nsw-rfs-incidents</li>
    <li>python311Packages.aio-geojson-usgs-earthquakes</li>
    <li>python311Packages.crate</li>
    <li>python311Packages.geojson</li>
    <li>python311Packages.geojson-client</li>
    <li>python311Packages.labelbox</li>
    <li>python311Packages.osmpythontools</li>
    <li>python311Packages.pyowm</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>csvkit</li>
    <li>home-assistant-component-tests.geonetnz_quakes</li>
    <li>home-assistant-component-tests.geonetnz_volcano</li>
    <li>home-assistant-component-tests.nsw_rural_fire_service_feed</li>
    <li>home-assistant-component-tests.usgs_earthquakes_feed</li>
    <li>python310Packages.agate-sql</li>
    <li>python310Packages.aio-geojson-client</li>
    <li>python310Packages.aio-geojson-geonetnz-quakes</li>
    <li>python310Packages.aio-geojson-geonetnz-volcano</li>
    <li>python310Packages.aio-geojson-nsw-rfs-incidents</li>
    <li>python310Packages.aio-geojson-usgs-earthquakes</li>
    <li>python310Packages.crate</li>
    <li>python310Packages.geojson</li>
    <li>python310Packages.geojson-client</li>
    <li>python310Packages.labelbox</li>
    <li>python310Packages.osmpythontools</li>
  </ul>
</details>

oxzi commented:
Thanks for running `nixpkgs-review`, @mattmelling.

I have only tested `python3`, not `python311` which is currently Python 3.11.1.
Due to a bug (?) in the project's `setup.py`, this version was not supported which I fixed through https://github.com/jazzband/geojson/pull/198.
I have just added a patch to this PR including this fix.

Now `python311packages.geojson` builds and I have just launched a `nixpkgs-review` job myself.

oxzi commented:
Result of `nixpkgs-review pr 214533` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>33 packages failed to build:</summary>
  <ul>
    <li>csvkit</li>
    <li>home-assistant-component-tests.geo_json_events</li>
    <li>home-assistant-component-tests.geonetnz_quakes</li>
    <li>home-assistant-component-tests.geonetnz_volcano</li>
    <li>home-assistant-component-tests.nsw_rural_fire_service_feed</li>
    <li>home-assistant-component-tests.openweathermap</li>
    <li>home-assistant-component-tests.usgs_earthquakes_feed</li>
    <li>python310Packages.agate-sql</li>
    <li>python310Packages.aio-geojson-client</li>
    <li>python310Packages.aio-geojson-generic-client</li>
    <li>python310Packages.aio-geojson-geonetnz-quakes</li>
    <li>python310Packages.aio-geojson-geonetnz-volcano</li>
    <li>python310Packages.aio-geojson-nsw-rfs-incidents</li>
    <li>python310Packages.aio-geojson-usgs-earthquakes</li>
    <li>python310Packages.crate</li>
    <li>python310Packages.geojson</li>
    <li>python310Packages.geojson-client</li>
    <li>python310Packages.labelbox</li>
    <li>python310Packages.osmpythontools</li>
    <li>python310Packages.pyowm</li>
    <li>python311Packages.agate-sql</li>
    <li>python311Packages.aio-geojson-client</li>
    <li>python311Packages.aio-geojson-generic-client</li>
    <li>python311Packages.aio-geojson-geonetnz-quakes</li>
    <li>python311Packages.aio-geojson-geonetnz-volcano</li>
    <li>python311Packages.aio-geojson-nsw-rfs-incidents</li>
    <li>python311Packages.aio-geojson-usgs-earthquakes</li>
    <li>python311Packages.crate</li>
    <li>python311Packages.geojson</li>
    <li>python311Packages.geojson-client</li>
    <li>python311Packages.labelbox</li>
    <li>python311Packages.osmpythontools</li>
    <li>python311Packages.pyowm</li>
  </ul>
</details>

oxzi commented:
Result of `nixpkgs-review pr 214533` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>home-assistant-component-tests.geo_json_events</li>
    <li>home-assistant-component-tests.openweathermap</li>
    <li>python310Packages.aio-geojson-generic-client</li>
    <li>python310Packages.pyowm</li>
    <li>python311Packages.aio-geojson-generic-client</li>
    <li>python311Packages.labelbox</li>
    <li>python311Packages.pyowm</li>
  </ul>
</details>
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>csvkit</li>
    <li>home-assistant-component-tests.geonetnz_quakes</li>
    <li>home-assistant-component-tests.geonetnz_volcano</li>
    <li>home-assistant-component-tests.nsw_rural_fire_service_feed</li>
    <li>home-assistant-component-tests.usgs_earthquakes_feed</li>
    <li>python310Packages.agate-sql</li>
    <li>python310Packages.aio-geojson-client</li>
    <li>python310Packages.aio-geojson-geonetnz-quakes</li>
    <li>python310Packages.aio-geojson-geonetnz-volcano</li>
    <li>python310Packages.aio-geojson-nsw-rfs-incidents</li>
    <li>python310Packages.aio-geojson-usgs-earthquakes</li>
    <li>python310Packages.crate</li>
    <li>python310Packages.geojson</li>
    <li>python310Packages.geojson-client</li>
    <li>python310Packages.labelbox</li>
    <li>python310Packages.osmpythontools</li>
    <li>python311Packages.agate-sql</li>
    <li>python311Packages.aio-geojson-client</li>
    <li>python311Packages.aio-geojson-geonetnz-quakes</li>
    <li>python311Packages.aio-geojson-geonetnz-volcano</li>
    <li>python311Packages.aio-geojson-nsw-rfs-incidents</li>
    <li>python311Packages.aio-geojson-usgs-earthquakes</li>
    <li>python311Packages.crate</li>
    <li>python311Packages.geojson</li>
    <li>python311Packages.geojson-client</li>
    <li>python311Packages.osmpythontools</li>
  </ul>
</details>

oxzi commented:
Now this looks better.. somehow `nixpkgs-review` gets very confused if a GC run interferes.

* `python311Packages.labelbox` already fails on the current `master` branch.
* `aio-geojson-generic-client` expects `geojson >= 2.4.0` and was thereby broken before.
* `pyowm` even demands `geojson >= 2.3.0`.
* `home-assistant-component-tests.geo_json_events` uses `aio-geojson-generic-client` and `home-assistant-component-tests.openweathermap` uses `python311Packages.pyowm`.

Thus, there don't seem to be any new breakages.

oxzi commented:
This PR became irrelevant after @mweinelt pushed e63354a73051709610567e986cb4976601237a3c.
